File: examples/java-repo/src/main/java/com/example/ecommerce/platform/config/AppConfig.java
package com.example.ecommerce.platform.config;

public class AppConfig {

    // DELIBERATE: mutable static field (SpotBugs MS_SHOULD_BE_FINAL)
    public static String APP_NAME = "ecommerce";
    public static String APP_VERSION = "1.0.0";

    private final String environment;

    public AppConfig(String environment) {
        if (environment == null || environment.isBlank()) {
            this.environment = "development";
        } else {
            this.environment = environment;
        }
    }

    public String getEnvironment() {
        return environment;
    }

    public boolean isProduction() {
        return "production".equals(environment);
    }

    public String getAppName() {
        return APP_NAME;
    }

    public String getAppVersion() {
        return APP_VERSION;
    }
}

File: examples/go-repo/internal/platform/config/app_config.go
package config

// deliberate gochecknoglobals: global mutable variable
var AppName = "ecommerce"
var AppVersion = "1.0.0"
var Debug = false

type AppConfig struct {
	Name    string
	Version string
	Debug   bool
}

func NewAppConfig() AppConfig {
	return AppConfig{
		Name:    AppName,
		Version: AppVersion,
		Debug:   Debug,
	}
}
